src/shared/components/search-result/search-result.component.ts

```diff
@@ -109,6 +121,72 @@ export class SearchResultComponent implements OnChanges {
     }
   }
 
+  public clearSelection(): void {
+    this.selectedCases = [];
+    this.selection.emit(this.selectedCases);
+  }
+
+  public canBeShared(caseView: CaseView): boolean {
+    return true;
+  }
+
+  public selectAll(): void {
+    if (this.allOnPageSelected()) {
+      // all cases already selected, so unselect all on this page
+      this.resultView.results.forEach(c => {
+        this.selectedCases.forEach((s, i) => {
+          if (c.case_id === s.case_id) {
+            delete this.selectedCases[i];
+          }
+        });
+      });
+    } else {
+      this.resultView.results.forEach(c => {
+        if (!this.isSelected(c)) {
+          this.selectedCases.push(c);
+        }
+      });
+    }
+    this.selection.emit(this.selectedCases);
+  }
+
+  public changeSelection(c: SearchResultViewItem): void {
```

Comment:
Should this function not be doing this.selection.emit(this.selectedCases); as well?

Reply:
Yes it should